runtime/cudaq/platform/default/rest_server/helpers/RestRemoteClient.cpp

```diff
@@ -136,11 +136,19 @@ class RemoteRestRuntimeClient : public cudaq::RemoteRuntimeClient {
       // Create a new Module to clone the function into
       auto location = FileLineColLoc::get(&mlirContext, "<builder>", 1, 1);
       ImplicitLocOpBuilder builder(location, &mlirContext);
+      // Add cuda quantum kernel attribute if not already set.
+      if (!func->hasAttr(cudaq::kernelAttrName))
+        func->setAttr(cudaq::kernelAttrName, builder.getUnitAttr());
       // Add entry-point attribute if not already set.
       if (!func->hasAttr(cudaq::entryPointAttrName))
         func->setAttr(cudaq::entryPointAttrName, builder.getUnitAttr());
       auto moduleOp = builder.create<ModuleOp>();
-      moduleOp.push_back(func.clone());
+      for (auto &op : *module) {
+        auto funcOp = dyn_cast<func::FuncOp>(op);
+        // Add quantum kernels defined in the module.
+        if (funcOp && funcOp->hasAttr(cudaq::kernelAttrName))
+          moduleOp.push_back(funcOp.clone());
```

Comment:
Double check me on this, but wouldn't we also want to allow classical calls defined in the kernel, too? (But only if they're truly defined, not just declared and hoped to be linked in later.)

Reply:
That's right. But the bridge would only process quantum kernels, thus only those related to them are present in the quake code. This filter is mainly for non-JIT wrappers (required for direct execution) such as `altLaunchKernel`, `thunk`, etc., which we don't need when invoking the JIT'ed function.

A side effect of those wrappers is that they are not compatible with the synthesizer pass since they call the original kernel signature, hence will fail IR verification.